src/coreclr/ilasm/writer.cpp

```diff
@@ -1289,7 +1458,31 @@ HRESULT Assembler::CreatePEFile(_In_ __nullterminated WCHAR *pwzOutputFilename)
 
     if (FAILED(hr=CreateTLSDirectory())) goto exit;
 
-    if (FAILED(hr=CreateDebugDirectory())) goto exit;
+    if (m_fGeneratePDB)
+    {
+        mdMethodDef entryPoint;
+
+        if (FAILED(hr = m_pCeeFileGen->GetEntryPoint(m_pCeeFile, &entryPoint))) goto exit;
+        if (FAILED(hr = m_pPortablePdbWriter->BuildPdbStream(m_pEmitter, entryPoint))) goto exit;
+
+        BYTE pdbChecksum[32];
+        if (FAILED(hr = m_pPortablePdbWriter->ComputeSha256PdbStreamChecksum(pdbChecksum))) goto exit;
+
+        if (m_fDeterministic)
+        {
+            // Get deterministic GUID and timestamp from the computed hash
+            _ASSERTE(sizeof(GUID) + sizeof(ULONG) <= sizeof(pdbChecksum));
+            GUID pdbGuid = *((GUID*)&pdbChecksum);
+            if (FAILED(hr = m_pPortablePdbWriter->ChangePdbStreamGuid(pdbGuid))) goto exit;
+
+            ULONG timestamp;
+            memcpy_s(&timestamp, sizeof(ULONG), pdbChecksum + sizeof(GUID), sizeof(ULONG));
+            m_pPortablePdbWriter->SetTimestamp(timestamp);
```

Comment:
Do we need to generate deterministic GUID from the content hash even when we are not generating the PDBs?

Reply:
That seems better than leaving the GUID/timestamp as zero. Are you suggesting we derive them from something like the hash we use for the MVID below, if the PDB stream isn't available?

Reply:
Yes. I believe that generating .dll without .pdbs is fairly common.

Reply:
It would be best to avoid depending on PDB stream to generate the GUID/timestamp in all cases, even when we are genering the PDBs.